safe_api.js

```diff
@@ -78,10 +133,41 @@ window.safeAPI = {};
 					.then(() => mdHandle));
 		}
 
+		getString(key) {
+			return this.then(_ => {
+				console.log('Fetching key from handle: ', key, this.mdHandle);
+				return safeMutableData.get(this.mdHandle, key);
+			})
+			.then((val) => {
+				return val.buf.toString();
+			});
+		}
+
 		free() {
-			this.handlePromise.then((mdHandle) => {
-				safeMutableData.free(mdHandle);
-				this.handlePromise = null;
+			this.then((mdHandle) => {
```

Comment:
You will get `appHandle` as `mdHandle` here, because `this.then` calls `this.parent.then` ?

Reply:
`mdHandle` depends entirely on whatever the last function returns. It might not even be an mdHandle. It might be null if the previous `thenable` called has a Promise that returns nothing.

That's why in the code I was using, I switched to `self.handle` in the code I was personally using and tested.

Reply:
Thanks, but I think having a cleaner code is more important than breaking existing one :-) So break everything you want.